pkg/parcacol/schema.go

```diff
@@ -6,68 +6,83 @@ import (
-			Dynamic:       false,
-		}, {
-			Name:          columnPeriodUnit,
-			StorageLayout: parquet.Encoded(parquet.String(), &parquet.RLEDictionary),
-			Dynamic:       false,
-		}, {
-			Name:          columnLabels,
-			StorageLayout: parquet.Encoded(parquet.Optional(parquet.String()), &parquet.RLEDictionary),
-			Dynamic:       true,
-		}, {
-			Name:          columnStacktrace,
-			StorageLayout: parquet.Encoded(parquet.String(), &parquet.RLEDictionary),
-			Dynamic:       false,
-		}, {
-			Name:          columnTimestamp,
-			StorageLayout: parquet.Int(64),
-			Dynamic:       false,
-		}, {
-			Name:          columnDuration,
-			StorageLayout: parquet.Int(64),
-			Dynamic:       false,
-		}, {
-			Name:          columnPeriod,
-			StorageLayout: parquet.Int(64),
-			Dynamic:       false,
-		}, {
-			Name:          columnValue,
-			StorageLayout: parquet.Int(64),
-			Dynamic:       false,
-		}},
+		[]dynparquet.ColumnDefinition{
+			{
+				Name:          columnDuration,
+				StorageLayout: parquet.Int(64),
+				Dynamic:       false,
+			}, {
+				Name:          columnLabels,
+				StorageLayout: parquet.Encoded(parquet.Optional(parquet.String()), &parquet.RLEDictionary),
+				Dynamic:       true,
+			}, {
+				Name:          columnPeriod,
+				StorageLayout: parquet.Int(64),
+				Dynamic:       false,
+			}, {
+				Name:          columnPeriodType,
+				StorageLayout: parquet.Encoded(parquet.String(), &parquet.RLEDictionary),
+				Dynamic:       false,
+			}, {
+				Name:          columnPeriodUnit,
+				StorageLayout: parquet.Encoded(parquet.String(), &parquet.RLEDictionary),
+				Dynamic:       false,
+			}, {
+				Name:          columnPprofLabels,
+				StorageLayout: parquet.Encoded(parquet.Optional(parquet.String()), &parquet.RLEDictionary),
+				Dynamic:       true,
+			}, {
+				Name:          columnPprofNumLabels,
+				StorageLayout: parquet.Encoded(parquet.Optional(parquet.String()), &parquet.RLEDictionary),
+				Dynamic:       true,
+			}, {
+				Name:          columnSampleType,
+				StorageLayout: parquet.Encoded(parquet.String(), &parquet.RLEDictionary),
+				Dynamic:       false,
+			}, {
+				Name:          columnSampleUnit,
+				StorageLayout: parquet.Encoded(parquet.String(), &parquet.RLEDictionary),
+				Dynamic:       false,
+			}, {
+				Name:          columnStacktrace,
+				StorageLayout: parquet.Encoded(parquet.String(), &parquet.RLEDictionary),
+				Dynamic:       false,
+			}, {
+				Name:          columnTimestamp,
+				StorageLayout: parquet.Int(64),
+				Dynamic:       false,
+			}, {
+				Name:          columnValue,
+				StorageLayout: parquet.Int(64),
+				Dynamic:       false,
+			},
+		},
 		[]dynparquet.SortingColumn{
 			dynparquet.Ascending(columnSampleType),
 			dynparquet.Ascending(columnSampleUnit),
 			dynparquet.Ascending(columnPeriodType),
 			dynparquet.Ascending(columnPeriodUnit),
+			dynparquet.NullsFirst(dynparquet.Ascending(columnPprofLabels)),
```

Comment:
I expected these to be after the stacktrace in their sorting. Potentially even after the timestamp (which is what was originally proposed in the RFC).